### PR TITLE
Add h5py deps as it is required for two unused post-processing functions

### PR DESCRIPTION
**Pull request by jorgensd**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -43,11 +43,11 @@
         'tabulate',
         'termcolor',
         'pytest',
-        'sphinx==4.3.0',
+        # 'sphinx==4.3.0',
```
Comment (jorgensd) [inline setup.py]:
```suggestion
        'sphinx==4.3.0',
```

**Comment by justinlaughlin** [comment]
Oops - my bad for not adding that dependency in. I used `h5py` quite a bit in actual simulations to work around the mesh partitioning issue; I think its relevant enough to be a core dependency. Thank you for adding to CI.

**Review by justinlaughlin — APPROVED** [review]
A noob question - but does the `--no-binary` flag mean we use the binary that comes with the base fenics container and `h5py` will wrap around that? In the dockerfile recipe I was using I ended up installing `python3-h5py` via `apt-get` because I couldn't figure out how to make things compatible. Very helpful.

**Comment by jorgensd** [comment]
> A noob question - but does the `--no-binary` flag mean we use the binary that comes with the base fenics container and `h5py` will wrap around that? In the dockerfile recipe I was using I ended up installing `python3-h5py` via `apt-get` because I couldn't figure out how to make things compatible. Very helpful.

Exactly. This is so we dont get incompatible HDF5 backenfs.

**Pull request merged**